ilastik/workflows/carving/carvingGui.py

```diff
@@ -245,33 +246,46 @@ def onSegmentButton(self):
         biasValue = self.labelingDrawerUi.noBiasBelowSpin.value()
         self.topLevelOperatorView.NoBiasBelow.setValue(biasValue)
         self.topLevelOperatorView.Trigger.setDirty(slice(None))
-    
+
+    def getObjectNames(self):
+        return self.topLevelOperatorView.AllObjectNames[:].wait()
+
     def saveAsDialog(self, name=""):
         '''special functionality: reject names given to other objects'''
+        namesInUse = self.getObjectNames()
+
+        def generateObjectName():
+            return "{} {}".format(self.objectPrefix, len(namesInUse) + 1)
```

Comment:
just a matter of taste, but I prefer the [F-string syntax](https://www.python.org/dev/peps/pep-0498/)...
```suggestion
            return f"{self.objectPrefix} {len(namesInUse) + 1}"
```